### DISCUSSION TITLE
Programmatically Close Dialog

### DISCUSSION THREAD

**Discussion by ColeMiller21** [shadcn-ui/ui#1504] [Q&A] [answered]
I have a dialog that will save changes - but I want it to close if the submit method is called successfully and unsure on how to accomplish this?

**Comment by joaom00** [comment, marked as answer]
To control the Dialog you can do this
```tsx
const [open, setOpen] = React.useState(false)

<Dialog open={open} onOpenChange={setOpen}>
  ...
</Dialog>
```

**Reply by ColeMiller21** [reply]
Thank you very much for the quick response!

**Reply by hamzabenarfa** [reply]
thnx

**Reply by ChibuzoEkwue** [reply]
Thanks

**Comment by Inaam-Ur-Rehman** [comment]
you can use react useRef hook.

` 
const ref = React.useRef();
`
Then in DialogTrigger:
`<DialogTrigger ref={ref}>`
Next, you can add this from where you want to close this dialog.
`ref.current?.click();`

**Comment by Psionyc** [comment]
This is the code I use to dismiss my dialog. I put it into a hook to makes things easier. It only works on client

```tsx
import React from 'react'

const useDismissModal = () => {
    const dismiss = () => (document.querySelector('[data-state="open"]') as HTMLDivElement).click()
    return {
        dismiss
    }
}

export default useDismissModal
```
